bigquery/google/cloud/bigquery/table.py

```diff
@@ -791,30 +791,40 @@ def keys(self):
         >>> Row(('a', 'b'), {'x': 0, 'y': 1}).keys()
         ['x', 'y']
         """
-        for k in self._xxx_field_to_index.keys():
-            yield k
+        keys = self._xxx_field_to_index.keys()
+        return keys
 
     def items(self):
         """
         Return items as of a dict:
         >>> Row(('a', 'b'), {'x': 0, 'y': 1}).items()
         [('x', 'a'), ('y', 'b')]
         """
-        for k, i in self._xxx_field_to_index.items():
-            yield k, self._xxx_values[i]
-
-    def dict(self):
-        """
-        Return row as a dict object:
-        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).dict()
-        {'x': 'a', 'y': 'b'}
-        """
-        data = {
-            k: self._xxx_values[i]
+        items = [
+            (k, self._xxx_values[i])
             for k, i
             in self._xxx_field_to_index.items()
-        }
-        return data
+        ]
+        return items
+
+    def get(self, key, default = None):
+        """
+        Return value under specified key
+        Defaults to None or specified default
+        if key does not exist:
+        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).get('x')
+        'a'
+        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).get('z')
+        None
+        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).get('z', '')
+        ''
+        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).get('z', default = '')
+        ''
+        """
+        index = self._xxx_field_to_index.get(key)
+        if index is None:
```

Comment:
I think this might not quite be right in the case where the key exists, but is explicitly set to None.

```
>>> d = {'x': None}
>>> y = d.get('x', 'default')
>>> print(y)
None
```

Reply:
Nevermind. I see this is checking if the index is None, not the value.